services/provider/server/consumer.go

```diff
@@ -224,3 +224,18 @@ func (c *ocsConsumerManager) Get(ctx context.Context, id string) (*ocsv1alpha1.S
 
 	return consumerObj, nil
 }
+
+func (c *ocsConsumerManager) UpdateConsumerReportedStatus(ctx context.Context, id string) error {
+	consumerObj, err := c.Get(ctx, id)
+	if err != nil {
+		return err
+	}
+	consumerObj.Status.LastHeartbeat = metav1.Now()
+	err = c.client.Status().Update(ctx, consumerObj)
```

Comment:
Updated in the latest commit
We are using patch because update will rewrite the entire object whereas patch will replace the intended field.
Another advantage of using patch is that it will save an API call to server as only the updated data needs to be provided.